lib/src/Base/Algo/KarhunenLoeveResultImplementation.cxx

```diff
@@ -152,10 +152,15 @@ Point KarhunenLoeveResultImplementation::project(const Function & function) cons
 Point KarhunenLoeveResultImplementation::project(const Field & field) const
 {
   if (field.getMesh() == modesAsProcessSample_.getMesh())
-    return projection_ * field.getValues().getImplementation()->getData();
+    return project(field.getValues());
   return project(Function(new P1LagrangeEvaluation(field)));
```

Comment:
I see why you kept project as is, but honestly it would be cleaner to state that `KLResult::project` projects values on its input mesh, and user would call P1LagrangeInterpolation if needed.